src/js/reducers/sessionReducer.jsx

```diff
@@ -121,6 +123,8 @@ export default function (state = initialState, action) {
         displayDateDefaultValue: _.get(action, 'payload.data.data.displayDateDefaultValue', '-'),
         notificationAutohideDelay: _.get(action, 'payload.data.data.notificationAutohideDelay', 8000),
         browserConnectionTimeout: _.get(action, 'payload.data.data.browserConnectionTimeout', 0),
+        isAutosaveEnabled: _.get(action, 'payload.data.data.isAutosaveEnabled') &&
+          _.includes(_.get(action, 'payload.data.data.supportedActivities'), ActivityCode.AUTOSAVE),
```

Comment:
I'd add a helper function outside a reduce to help readability:
```javascript
const isAutosaveEnabled = (payload) => {
   const { isAutosaveEnabled, supportedActivities } = payload?.data?.data
   return isAutosaveEnabled && supportedActivities.includes(ActivityCode.AUTOSAVE);
}
```

and in the reducer just use:
```javascript
isAutosaveEnabled: isAutosaveEnabled(action.payload)
```